src/coreclr/src/jit/simdcodegenxarch.cpp

```diff
@@ -2192,8 +2192,12 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
     {
         unsigned simdInitTempVarNum = compiler->lvaSIMDInitTempVarNum;
         noway_assert(simdInitTempVarNum != BAD_VAR_NUM);
-        bool      isEBPbased;
-        unsigned  offs     = compiler->lvaFrameAddress(simdInitTempVarNum, &isEBPbased);
+        bool     isEBPbased;
+        unsigned offs = compiler->lvaFrameAddress(simdInitTempVarNum, &isEBPbased);
+
+        // Adjust the offset with stack level, if any.
```

Comment:
This could use a comment. I generally find it confusing to reason about the stack level, but I believe it's the case that 1) it will always be zero if there's a frame pointer (I was wondering why you weren't checking `isEBPbased`), and 2) it will always be zero when `FEATURE_FIXED_OUT_ARGS` is true. Can you confirm or correct?
Also, I'm wondering why this isn't needed further down when a temp is used.

Reply:
Normally, all local access is passed down to the emitter by storing the local number and offset within the local, and `emitOutputSV` does the appropriate offset translation using:
```
#if !FEATURE_FIXED_OUT_ARGS
        // Adjust the offset by the amount currently pushed on the CPU stack
        dsp += emitCurStackLvl;
#endif
```

It looks like the two cases that don't do this are the calls to `lvaFrameAddress` in this function, `genSIMDIntrinsicGetItem`.

So it seems like this is ok, but you need the `#if !FEATURE_FIXED_OUT_ARGS` case (for Linux x86 ABI) or prove Carol's #2. It would probably be better to only do the adjustment for the `!isEBPbased` case, for clarity.

Also, I think the op1 case above that calls `lvaFrameAddress` also needs this logic.

Reply:
Alright, so just verified that for `isEBPbased == true`, `genStackLevel` can be non-zero. I was hitting previously added assert when crossgening `System.Numerics.Plane:CreateFromVertices`. I have now updated the assert to only include when `FEATURE_FIXED_OUT_ARGS` is `true`.  Below is the code that was getting generated when I hit the assert:

```asm
       83EC0C       sub      esp, 12
       F20F110424   movsd    qword ptr [esp], xmm0
       660F70C802   pshufd   xmm1, xmm0, 2
       F30F114C2408 movss    dword ptr [esp+08H], xmm1
       F30F108568FFFFFF movss    xmm0, dword ptr [ebp-98H]  ; <-- This line
```
In above highlighted line, offset returned by `lvaFrameAddress` was 0x98 and genStackLevel was 0n12 inside `genSimdIntrinsicGetItem`. Since it was `isEBPBased`, the stack level didn't get added to the offset 0x98 and everything now works fine. So what I can conclude is for EBP based, the `varDsc->lvStkOff` holds accurate value but that doesn't happen when ESP based?

Reply:
> So what I can conclude is for EBP based, the varDsc->lvStkOff holds accurate value but that doesn't happen when ESP based?

Yes, that's right - because `genStackLevel` is relative to SP. EBP remains fixed, so the offset relative to that doesn't change.